zenodo/modules/auditor/oai.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2016 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Zenodo Auditor for OAI-PMH."""
+
+from itertools import chain
+
+import sqlalchemy as sa
+from elasticsearch_dsl import Q
+from flask import current_app
+from invenio_communities.models import Community
+from invenio_db import db
+from invenio_oaiserver.models import OAISet
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+from invenio_records.models import RecordMetadata
+from invenio_search import RecordsSearch
+from mock import patch
+from redis import StrictRedis
+from sickle import Sickle
+from sqlalchemy import cast
+
+from .api import Audit, Check
+from .utils import sickle_requests_get_mock
+
+
+class OAIAudit(Audit):
+    """OAI Audit."""
+
+    def __init__(self, audit_id, logger, communities):
+        """Initialize an OAI audit."""
+        super(OAIAudit, self).__init__(audit_id, logger)
+
+        self._build_oai_sets_from_db()
+        self._checks = chain(iter((OAICorrespondenceCheck(),)),
+                             (OAISetResultCheck(self, c) for c in communities))
+
+    def _build_oai_sets_from_db(self):
+        self.redis = StrictRedis()
```

Comment:
:+1: I didn't notice that it was a generic Flask-Cache. I'm using Redis sets though, so I'll have to assert Redis for the `CACHE_TYPE`.